lib/rules/linebreak-style.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @fileoverview Rule to forbid mixing LF and LFCR line breaks.
+ * @author Erik Mueller
+ * @copyright 2015 Erik Mueller. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    var linebreakStyle = context.options[0] || "unix";
+
+    return {
+        "Program": function checkForlinebreakStyle(node) {
+            var source = context.getSource(),
+                expectedLF = linebreakStyle === "unix",
+                // Assume LF if we find not one CRLF
+                isSolelyLF = source.indexOf("\r\n") === -1,
+                // Assume CRLF if we find no LF
+                isSolelyCRLF = source.match(/[^\r]\n/) === null;
+
+            if (expectedLF && isSolelyCRLF) {
+                context.report(node, "Expected linebreaks to be 'LF' but found 'CRLF'.");
+            } else if (!expectedLF && isSolelyLF) {
+                context.report(node, "Expected linebreaks to be 'CRLF' but found 'LF'.");
+            } else if (!isSolelyLF && !isSolelyCRLF) {
+                // NOTE How to tell the line where the error occured when only having the complete source code?
```

Comment:
Perhaps use the regex from the `getSourceLines()` implementation as a capture group on the whole source, then iteratively make sure each match is valid according to the expected option? Line would be the number of groups matched so far, column the length of the line in `getSourceLines()`.